### PR TITLE
fix(TFD-3854): Ensure that top-level generated records are named.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**What is the current behavior?** (You can also link to an open issue here)

Avro requires top-level records to be named.  The KV utilities can generate records without names.

**What is the new behavior?**

All empty records use the single definition provided by AvroUtil.
If the key or value records are unnamed, they are given a default name.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java [hunk 1]
```diff
@@ -204,15 +210,18 @@ public static Schema convertTreeToAvroSchema(Map<String, Set<Object>> tree, Stri
...
-                    inputSchema.getNamespace(), inputSchema.isError(), new ArrayList<Schema.Field>());
-        } else {
-            return outputSchema;
+        Schema schema = extractValues(inputSchema, keyPaths, "");
+        if (schema == null) {
+            return Schema.createRecord("value_" + inputSchema.getName(), inputSchema.getDoc(), inputSchema.getNamespace(),
+                    inputSchema.isError(), new ArrayList<Schema.Field>());
+        }
+        if (schema.getName() == null) {
+            schema = Schema.createRecord(RECORD_VALUE_PREFIX, schema.getDoc(), schema.getNamespace(), schema.isError(),
```
Comment (pradasouvanlasy) [inline core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java]: I don't get why the chosen name should be different between this case and the one above (when `schema == null`). Does that matter ?

Comment (RyanSkraba) [reply core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java]: Good point!  Will fix

File: core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java [hunk 2]
```diff
@@ -204,15 +210,17 @@ public static Schema convertTreeToAvroSchema(Map<String, Set<Object>> tree, Stri
...
-            return Schema.createRecord("value_" + inputSchema.getName(), inputSchema.getDoc(),
-                    inputSchema.getNamespace(), inputSchema.isError(), new ArrayList<Schema.Field>());
-        } else {
-            return outputSchema;
+        Schema schema = extractValues(inputSchema, keyPaths, "");
+        if (schema == null) {
+            return AvroUtils.createEmptySchema();
+        }
+        if (schema.getName() == null) {
+            schema = Schema.createRecord(RECORD_VALUE_PREFIX, schema.getDoc(), schema.getNamespace(), schema.isError(),
```
Comment (sgandon) [inline core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java]: is there any test testing this ?

Comment (pradasouvanlasy) [reply core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java]: It seems that this case is covered in `SchemaGeneratorUtilsTest`

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1829/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1829/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arskraba%2Ffix%2FTFD-3854-name-kv-schemas)

File: core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java (hunk 2 above)
Comment (sgandon) [reply core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/kv/SchemaGeneratorUtils.java]: thanks

**Pull request merged**